src/quo2/components/wallet/transaction_progress/view.cljs

```diff
@@ -0,0 +1,306 @@
+    (or (= network-state :confirmed) (= network-state :finalising)) "Confirmed on"
+    (= network-state :finalized)                                    "Finalized on"
+    (= network-state :error)                                        "Failed on"))
+
+(defn steps-text
+  [network-type network-state]
+  (cond
+    (and (= network-type :mainnet)
+         (not= network-state :finalized)
+         (not= network-state :error))       (str (if (< (@app-state :counter) 4)
+                                                   (@app-state :counter)
+                                                   "4")
+                                                 "/4")
+    (= network-state :finalized)            "Epoch 181,329"
+    (and (= network-type :mainnet)
+         (= network-state :error))          "0/4"
+    (and (not= network-type :mainnet)
+         (or (= network-state :finalising)
+             (= network-state :confirmed))) "1/1"
+    (and (= network-type :arbitrum)
+         (= network-state :sending))        "1/1"
+    (not= network-type :mainnet)            "0/1"))
+
+(defn get-status-icon
+  [network-type network-state]
+  (cond
+    (and (= network-type :arbitrum)
+         (= network-state :sending))   ["positive-state"
+                                        (colors/theme-colors colors/white
+                                                             colors/success-50
+                                                             colors/success-60)]
+    (or (= network-state :pending)
+        (= network-state :sending))    ["pending-state"
+                                        (colors/theme-colors colors/white
+                                                             colors/neutral-50
+                                                             colors/neutral-60)]
+    (or (= network-state :confirmed)
+        (= network-state :finalising)) ["positive-state"
+                                        (colors/theme-colors colors/white
+                                                             colors/success-50
+                                                             colors/success-60)]
+    (= network-state :finalized)       ["diamond"
+                                        (colors/theme-colors colors/white
+                                                             colors/success-50
+                                                             colors/success-60)]
+    (= network-state :error)           ["negative-state"
+                                        (colors/theme-colors colors/danger
+                                                             colors/danger-50
+                                                             colors/danger-60)]))
+
+(defn transaction-progress
+  [{:keys [title
+           on-press
+           accessibility-label
+           network-type
+           network-state
+           start-interval-now
+           override-theme]}]
```

Comment:
Quick one: the name we've been trying to use is `theme` instead of `override-theme`.